notebooks/Watermark photo attachments with Exif data.ipynb

```diff
@@ -0,0 +1,179 @@
+    "- Pillow Library - https://pillow.readthedocs.io/en/stable/index.html"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import shutil\n",
+    "import tempfile\n",
+    "from arcgis.gis import GIS\n",
+    "from datetime import datetime as dt\n",
+    "from PIL import ExifTags\n",
+    "from PIL import Image\n",
+    "from PIL import ImageDraw\n",
+    "from PIL import ImageOps\n",
+    "\n",
+    "\n",
+    "# Using built-in user\n",
+    "# For information on different authentication schemes see\n",
+    "# https://developers.arcgis.com/python/guide/working-with-different-authentication-schemes/\n",
+    "# and for protecting your credentials see\n",
+    "# https://developers.arcgis.com/python/guide/working-with-different-authentication-schemes/#protecting-your-credentials\n",
+    "gis = GIS(\"home\")\n",
+    "\n",
+    "# The item id of the feature layer to process attachments for\n",
+    "feature_layer_item_id = \"7a28f3f872bd4b55bfd7965bf73a46b6\"\n",
+    "# The id of the layer to use\n",
+    "feature_layer_id = 0\n",
+    "# Where clause to filter features and attachments\n",
+    "# See FeatureLayer query() for more options - https://developers.arcgis.com/python/api-reference/arcgis.features.toc.html#featurelayer\n",
+    "attachments_query = \"field_to_filter=2\"\n",
+    "\n",
+    "\n",
+    "# Converts latitude and longitude from Exif GPS info in degrees, minutes and seconds to decimal degrees\n",
+    "def convert_dms_to_dd(gps_info) -> (float, float):\n",
+    "\n",
+    "    if gps_info is None:\n",
+    "        return (0, 0)\n",
+    "\n",
+    "    lat_degrees = gps_info[ExifTags.GPS.GPSLatitude][0]\n",
+    "    lat_minutes = gps_info[ExifTags.GPS.GPSLatitude][1]\n",
+    "    lat_seconds = gps_info[ExifTags.GPS.GPSLatitude][2]\n",
+    "    lon_degrees = gps_info[ExifTags.GPS.GPSLongitude][0]\n",
+    "    lon_minutes = gps_info[ExifTags.GPS.GPSLongitude][1]\n",
+    "    lon_seconds = gps_info[ExifTags.GPS.GPSLongitude][2]\n",
+    "\n",
+    "    lat = float(lat_degrees + lat_minutes / 60 + lat_seconds / 3600)\n",
+    "    lon = float(lon_degrees + lon_minutes / 60 + lon_seconds / 3600)\n",
+    "\n",
+    "    if gps_info[ExifTags.GPS.GPSLatitudeRef] == \"S\":\n",
+    "        lat = -lat\n",
+    "    if gps_info[ExifTags.GPS.GPSLongitudeRef] == \"W\":\n",
+    "        lon = -lon\n",
+    "\n",
+    "    return (lat, lon)\n",
+    "\n",
+    "\n",
+    "# Watermarks an image with decimal degree latitude and longitude from Exif data\n",
+    "def add_dd_watermark(image_path):\n",
+    "\n",
+    "    image = Image.open(image_path)\n",
+    "    image = ImageOps.exif_transpose(image)\n",
+    "\n",
+    "    # Get the image's Exif data, convernt the lat and long to decmial degree format and formate the string for the watermark\n",
```

Comment:
Is this typo?
convernt--> convert
formate --> format

Reply:
Yes, and decmial
All kinds of typos in that line and I thought that I has spell checked this :(